app/controllers/markets_controller.rb

```diff
@@ -3,12 +3,48 @@ def home
 
   end
 
+  def select
+    redirect_to mkt_home_path(params[:market][:id])
+  end
+
+  def mkt_home
+    @market = Market.find(params[:id])
```

Comment:
Because we're setting the `@market` variable in this same way in a few different methods, I think we could DRY this out a bit by moving that line into a `before_action`, like so:
```ruby
class MarketsController < ApplicationController
before_action :find_market, only: [:mkt_home, :show, :edit, :update]

# all of your existing methods

private

  def find_market
    @market = Market.find(params[:id])
  end
end
```